account-kit/smart-contracts/src/light-account/clients/client.ts

```diff
@@ -94,28 +105,35 @@ export async function createLightAccountClient(
 ): Promise<SmartAccountClient | AlchemySmartAccountClient> {
   const { transport, chain } = params;
 
+  const account = await createLightAccount({
+    ...params,
+    transport,
+    chain,
+  });
   if (isAlchemyTransport(transport, chain)) {
-    return await createLightAccountAlchemyClient({
+    const account = await createLightAccount({
       ...params,
       transport,
+      chain,
     });
-  }
 
-  const lightAccount = await createLightAccount({
-    ...params,
-    transport,
-    chain,
-  });
+    return createAlchemySmartAccountClient({
+      ...params,
+      transport,
+      chain,
+      account,
+    }).extend(lightAccountClientActions);
+  }
 
   return createSmartAccountClient({
     ...params,
     transport,
-    chain: chain,
-    account: lightAccount,
+    chain,
+    account,
   }).extend(lightAccountClientActions);
 }
 
-function isAlchemyTransport(
+export function isAlchemyTransport(
```

Comment:
nit: this function should probably be exported from the same package that defines the alchemy transport (I think it's in infra?)